### DISCUSSION TITLE
How to perform `git push -u origin HEAD` easily in Magit?

### DISCUSSION THREAD

**Discussion by erikmd** [magit/magit#4416] [Support] [answered]
When branching, I typically want to push the new local branch to the same branch name (not already existing) in a given remote.

As documented in https://github.com/k88hudson/git-flight-rules#i-want-to-create-a-new-remote-branch-from-current-local-one,
this is easy doable in command-line: **`git push -u origin HEAD`** (assuming I set `git config --global push.default upstream`)

Is there a standard way to do this in Magit? or should Magit be slightly extended to do this feature in very few keystrokes?

as it's quite the last remaining command in my workflow that I still run in the terminal (outside of Magit :)

FTR, if my new local branch is called `feat-foo`, when I do <kbd>M-x magit-status RET P u</kbd>, I get:
`Set upstream of develop and push there (default origin/master):`
while I'd like to have by default something like:
`Set upstream of develop and push there (default origin/feat-foo):`
or
`Set upstream of develop and push there (default upstream/feat-foo):`

Cc @tarsius (hopefully you don't mind if I ping you :sweat_smile:, anyway I didn't create an issue, given this feature might just as well be already doable in Magit :)

**Comment by tarsius** [comment, marked as answer]
Try `P p`. It doesn't run that git command but accomplishes something similar.

Also note that it is useful to associate two other branches with a given local branch, not just one. And IMO you are using the upstream "wrong". See https://magit.vc/manual/3.0.0/magit/The-Two-Remotes.html#The-Two-Remotes.
